### PR TITLE
Mark or unmark a Product as Downloadable from Product Settings

### LINKED ISSUE #2757: Add/Edit Products M5: Downloadable files Step 2.1 - Mark/unmark a product as downloadable
As part of the M5 release, we need to provide the option to change the `is_downloadable` flag of a product by adding a checkbox "Product Downloadable" to the product settings:

iOS-product-settings

### PR DESCRIPTION
**Pull request by pmusolino**
Fixes #2757  

## Description
Now it's possible to mark/unmark every simple product as downloadable, from the Product Settings. This is part of Products MIlestone 5.

## Changes
- Introduced `isEditProductsRelease5Enabled` when needed for hiding the downloadable row in settings if the feature flag is not enabled.
- Added `DownloadableProduct` in `ProductSettingsRows`.
- Managed the appearance of the downloadable row in `ProductSettingsSections`.
- Added `downloadable` in `ProductSettings`.
- Added and updated unit tests.

#### Case 1
1. From the Products tab, select any Simple product.
2. Tap on the top right navigation bar button, then open the "Product Settings".
3. You should be able to see a new switchable row, titled `Downloadable Product`.
4. Make any changes on/off to the downloadable switch and update the product.
5. Come back again to this page to verify that your changes are reflected properly.

#### Case 2
1. From the Products tab, select a non-simple product.
2. Go to the product settings -> the Downloadable Product row shouldn't be visible.

## Screenshots
| Dark mode OFF             |  Dark mode ON |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 11 Pro - 2020-10-15 at 15 23 06 | Simulator Screen Shot - iPhone 11 Pro - 2020-10-15 at 15 24 58

## GIF

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: Do you think it's worth it to inject `ServiceLocator.featureFlagService` in the init of this view model(and set it as a default parameter), so it's clear that this view model has that dependency?

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift]: Should this go into a `Localization` enum as with other classes? 
I thought that was the standard on the codebase but correct me if I'm wrong!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift]: nit: extra space here

**Review by Ecarrion — APPROVED** [review]
Looks good! 
Just left a couple of suggestions for your consideration

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/ProductSettingsRows.swift]: Good idea. I moved all the `NSLocalizationString` of this class under `Localization` enum b71e516738260c44b7d775f455c3f5f7438bafbe

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift
Comment (pmusolino) [reply WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: Sounds good 👍 done here 027119293fe8f2fc5f4d26f1a0adb66e9bdd34eb

**Comment by pmusolino** [comment]
Thank you @Ecarrion for your review. 🙇 I'll wait for the @Garance91540 review before merging it.

**Review by Garance91540 — APPROVED** [review]
👏

**Comment by pmusolino** [comment]
Thank you both! Merging it now. <3

**Pull request merged**
